silo-core/contracts/utils/liquidationHelper/LiquidationHelper.sol

```diff
@@ -0,0 +1,147 @@
+        address payable _tokensReceiver
+    ) DexSwap(_exchangeProxy) {
+        NATIVE_TOKEN = _nativeToken;
+        EXCHANGE_PROXY = _exchangeProxy;
+        TOKENS_RECEIVER = _tokensReceiver;
+    }
+
+    /// @inheritdoc ILiquidationHelper
+    /// @dev entry point for liquidation
+    /// @notice for now we does not support liquidation with sTokens.
+    /// On not profitable liquidation we will revert, because we will not be able to repay flashloan with fee
+    /// (collateral will not be enough to cover loan + fee)
+    function executeLiquidation(
+        ISilo _flashLoanFrom,
+        address _debtAsset,
+        uint256 _maxDebtToCover,
+        LiquidationData calldata _liquidation,
+        DexSwapInput[] calldata _swapsInputs0x
+    ) external returns (uint256 withdrawCollateral, uint256 repayDebtAssets) {
+        require(_maxDebtToCover != 0, NoDebtToCover());
+
+        _flashLoanFrom.flashLoan(this, _debtAsset, _maxDebtToCover, abi.encode(_liquidation, _swapsInputs0x));
+        IERC20(_debtAsset).approve(address(_flashLoanFrom), 0);
+
+        withdrawCollateral = _withdrawCollateral;
+        repayDebtAssets = _repayDebtAssets;
+    }
+
+    /// @inheritdoc IERC3156FlashBorrower
+    function onFlashLoan(
+        address /* _initiator */,
+        address _debtAsset,
+        uint256 _maxDebtToCover,
+        uint256 _fee,
+        bytes calldata _data
+    )
+        external
+        returns (bytes32)
+    {
+        (
+            LiquidationData memory _liquidation,
+            DexSwapInput[] memory _swapInputs
+        ) = abi.decode(_data, (LiquidationData, DexSwapInput[]));
+
+        IERC20(_debtAsset).approve(address(_liquidation.hook), _maxDebtToCover);
+
+        (
+            _withdrawCollateral, _repayDebtAssets
+        ) = _liquidation.hook.liquidationCall({
+            _collateralAsset: _liquidation.collateralAsset,
+            _debtAsset: _debtAsset,
+            _user: _liquidation.user,
+            _maxDebtToCover: _maxDebtToCover,
+            _receiveSToken: false
+        });
+
+        IERC20(_debtAsset).approve(address(_liquidation.hook), 0);
+        uint256 flashLoanWithFee = _maxDebtToCover + _fee;
+
+        if (_liquidation.collateralAsset == _debtAsset) {
+            uint256 balance = IERC20(_liquidation.collateralAsset).balanceOf(address(this));
+            // bad debt is not supported, we will get underflow on bad debt
+            _transferToReceiver(_liquidation.collateralAsset, balance - flashLoanWithFee);
+        } else {
+            uint256 debtBalance = IERC20(_debtAsset).balanceOf(address(this));
+
+            if (flashLoanWithFee < debtBalance) {
+                unchecked {
+                    // safe because of `if (flashLoanWithFee < debtBalance)`
+                    _transferToReceiver(_debtAsset, debtBalance - flashLoanWithFee);
+                }
+            }
+
+            _executeSwap(_swapInputs); // part of collateral will be swap for (_maxDebtToCover + _fee)
```

Comment:
```suggestion
            _executeSwap(_swapInputs); // part of collateral will be swapped for (_maxDebtToCover + _fee)
```

What if swap will leave some leftover of debtAsset? I think it would be better to just transfer entire balances of collateral and debt assets at the end (minus flashloan repayment amount).

Reply:
few lines above we transferring debt (if there is anything left), on swap we want exact amount out == `flashLoan + fee`. Swap should not leave anything if we asking for `amount out`. Also, I think collateral is token we want to hold (at least that's most common case), not debt one, right?

Reply:
I think it's usually debt that is more valuable.

Reply:
ok, in that case I will change the flow

Reply:
with this approach, we will leave dust behind, but I see no way to handle it, I left comment in code.